Simplenote/src/main/java/com/automattic/simplenote/ThemedAppCompatActivity.java

```diff
@@ -51,6 +51,9 @@ public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, Strin
     public void recreate() {
         Intent intent = new Intent(this, getClass());
         intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        if (getIntent().getExtras() != null) {
+            intent.putExtras(getIntent().getExtras());
```

Comment:
We need to pass the bundle of current intent before recreating the activity, I missed this on the theme changes PR, and it caused a crash when the NoteEditorActivity was recreated to fix the theme.

Reply:
How do you reproduce the crash before these changes?  I wasn't able to get the app to crash after opening a note from the list or from a widget and toggling the theme on a small device.

Reply:
Sorry I wasn't clear enough 🙂, so first to be clear, I should've included this code by default, as when recreating the activity we need to pass the extras that we had before to avoid any unexpected behavior, and Android does this by default when we call the default `recreate` implementation: https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/app/ActivityThread.java;l=5419

In case we don't do this, when recreating `NoteEditorActivity` for example, we will have a null noteId here: https://github.com/Automattic/simplenote-android/blob/d43aeb4710c41a5f943430034b6cbcb0fe6cd8f5/Simplenote/src/main/java/com/automattic/simplenote/NoteEditorActivity.java#L101.
To reproduce the crash, pull the changes of this PR, then remove this change on `ThemedAppCompatActivity`, and try the exact steps described in the PR description, the app should crash after clicking the note in the widget.